include/s3select_functions.h

```diff
@@ -972,6 +974,51 @@ struct _fn_upper : public base_function {
     }
 };
 
+struct _fn_nullif : public base_function {
+
+    value x;
+    value y;
+
+    bool operator()(bs_stmt_vec_t* args, variable* result)
+    {
+        bs_stmt_vec_t::iterator iter = args->begin();
+
+        int args_size = args->size();
+        if (args_size != 2)
+        {
+          throw base_s3select_exception("nullif accept only 2 arguments");
+        }
+        base_statement *first = *iter;
+        x = first->eval();
+        iter++;
+        base_statement *second = *iter;
+        y = second->eval();
+        if (x.is_null() && y.is_null())
+        {
+          result->set_null();
+          return true;
+        }
+        if (x.is_null())
+        {
+          result->set_null();
+          return true;
+        }
+        if (!(x.is_number() && y.is_number()))
+        {
+          if (x.type != y.type)
```

Comment:
according to [documentation](https://docs.aws.amazon.com/AmazonS3/latest/dev/s3-glacier-select-sql-reference-conditional.html#s3-glacier-select-sql-reference-nullif)
there is no error in case of types are not equal.
for one example nullif(5,null) should return 5.

review the examples in the attach AWS documentation.
in other SQL engines (such as oracle), the describe behavior is different. 
currently, we implement AWS behavior.

Reply:
ew, ok